app/scripts/modules/core/src/presentation/modal/TabBoundary.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { useEventListener } from 'core';
+
+export const TabBoundary = ({ children }: { children: React.ReactNode }) => {
+  const dialogRef = React.createRef<HTMLDivElement>();
+
+  const keydownCallback = (event: KeyboardEvent) => {
+    const { keyCode, shiftKey, target } = event;
+    if (keyCode === 9) {
+      let targetOverride = null;
+      // tabbable selector via https://kennethbi.com/posts/5
+      const tabbableElements: HTMLElement[] = Array.from(
+        dialogRef.current.querySelectorAll(`
+          input:not([tabindex^="-"]):not([disabled]),
+          select:not([tabindex^="-"]):not([disabled]),
+          textarea:not([tabindex^="-"]):not([disabled]),
+          button:not([tabindex^="-"]):not([disabled]),
+          a[href]:not([tabindex^="-"]):not([disabled]),
+          [tabindex]:not([tabindex^="-"]):not([disabled])
+      `) as NodeListOf<HTMLElement>,
+      ).filter((ele: HTMLElement) => ele.offsetParent !== null);
+      if (!tabbableElements.length) {
```

Comment:
does this mean that if there are no tabbable elements inside `TabBoundary` we will allow focus to jump out of the boundary? If so, do we want that or should we also call `preventDefault()` here?

Reply:
That is what would happen. 

It should never happen with modals, because there's always the close button, but I don't know what we should be doing if this is used in other scenarios. It feels like it might be wrong to basically disable the tab key, but I...I dunno.

Reply:
🤷‍♂ Deciding is hard. I don't really know if I feel that bad about it in either direction - at the point where you don't have anything focusable living inside a component whose whole job is to deal with focus you've probably already gotten yourself into a weird situation.